### ISSUE TITLE
TeX Live インストール手順の解説

### ISSUE THREAD

**Issue by aminophen** [texjporg/texjporg.github.io#2]
しばらく止まってしまいましたが，公式ドキュメントを TeX Live 2019 リリースに合わせて公表できるようにしたいのですが，いかがでしょうか。扱う内容については #1 で議論中ですが，部分集合として

- 公式インストーラによるインストール方法
    - 例えば，（統合環境などのセットアップは扱わないという前提なので，）ターミナルから，TeX Wiki の「LaTeX 入門」の「最初の例」がターミナルからコンパイルできる状態になるまで

を書くことは異論ないと思います。これを最低限として，完成させたいです。

- 私は macOS と Windows は知っているので，時間を見つけて書こうと思います。（macOS → Windows の順で書くので，その前に Windows な方がいたら書いていただいて構いません。）
- Linux はどなたかお願いします。

**Comment by kmaed** [comment]
止まっていてすみません．Linux のはなんとか書きたいです．

**Comment by kmaed** [comment]
ようやく，だいぶざっくりですが，インストール終了までは書きました．とはいっても，まだ 2019 のイメージは CTAN にもないようなので動作未確認ですが．昔の記憶で，とりあえず ISO イメージ推奨な感じで書いています．

**Comment by aminophen** [comment]
@kmaed ありがとうございます。

ごめんなさい，私の方は時間がとれず全然進んでいません。2019 は GUI インストーラが新しくなっていますので，確認が遅れています。（Windows と Mac (MacTeX) は基本的にほとんどの人が GUI でインストールしているでしょうし…。）

**Comment by kmaed** [comment]
ようやく ISO イメージを落とせたので試してみました．Linux については従来通り CUI がデフォルトのようです．GUI は Tcl/Tk が必要で，ない場合は動かないです．CUI だと英語しか出ないですが，GUI にすると日本語にもできるようです．

**Comment by aminophen** [comment]
ネット環境が貧弱で全然試せていなかったのですが，ようやく ISO イメージを落とすことに成功しました。今から Windows 10 でインストールを試します。

疑問なのですが，ISO イメージに収録されているバイナリのプラットフォームは

* win32
* x86_64-darwin
* x86_64-darwinlegacy
* x86_64-cygwin
* x86_64-linux

だけでした。確かに，「DVD の容量不足のためいくつか省く」とはきいていたのですが，省かれたプラットフォームではどうやるのが良いのでしょう？

- ネットワークインストールしか手段はない？
- ISO イメージからパッケージだけインストールしておいて，後でバイナリだけインストールできる？それとも逆順で，先にバイナリだけインストールしておく？

**Comment by kmaed** [comment]
何も試していませんが，[Supported systems](https://www.tug.org/texlive/doc.html) には

> (*) Marked platforms are omitted from the DVD, but are available as usual via a network install.

とあります．例えば FreeBSD で ISO のインストーラを起動するとどうなるのでしょうね．

**Comment by kmaed** [comment]
もしかしたら i386-cygwin の方が試しやすいかも．

**Comment by aminophen** [comment]
Cygwin よくわからず諦めたことがあるので…。

FreeBSD 12.0 はこの前 VirtualBox にインストールしたんですが，GUI を使えるようにする時点でつまずいていて，今度は Solaris 11 ならどうにかなるんじゃないかと構築を試し始めました…。

**Comment by kmaed** [comment]
FreeBSD の X はどの辺でつまずきましたかね．GNOME などデスクトップ環境をちゃんと入れようとすると大変な気がします．私は試したことないですが，LXDE とかの方が楽なんじゃないでしょうか．いずれにせよ，知識がないとなかなか難しいと思います．

ISO のインストーラを試すだけならば，インストール直後の CUI でも十分じゃないかと思います．GUI インストーラまで試したいなら頑張らないとだめですが，どうせ Tk なので Linux などのと挙動は大して変わらないはずです．

**Comment by kmaed** [comment]
でも，インストール後にある程度使ってみるところまでやりたければ，確かに X は必要ですね……．

**Comment by aminophen** [comment]
確かに X がなくても iso からのインストールを試すだけなら出来ますね。常用するつもりはないので，あまり気張らず取っつきやすい方でやってみます。

（本題から逸れますが，FreeBSD を入れた理由は https://tug.org/pipermail/tlbuild/2019q2/004507.html のスレッドです。「BibTeXu が macOS，Windows，Linux では正常動作するのに FreeBSD, NetBSD, OpenBSD と Solaris で失敗する」らしい → 手元で試したいので FreeBSD 入れてみた → TeX Live のビルドまでは漕ぎ着けたが，デバッグには少なくとも vi か X が欲しい → でもどっちも良く知らないので挫折 → じゃあ代わりに Solaris にしよう…という感じで今に至ります）

**Comment by kmaed** [comment]
ICU の環境依存な違いで Unicode のソート結果がおかしくなるということでしょうか．検証は大変ですね……．

Solaris は使ったことがないので全くわかりませんが，X もバイナリパッケージで一発なんですかね．

GUI 環境がすぐ作れればよいですが，CUI で頑張る場合は……．エディタ何をお使いかわかりませんが，vi 使いならば何もしなくても単に vi と打てば起動するんじゃないでしょうか．vi 使いじゃない場合は，軽量のふつーのエディタとして nano とかを入れればとりあえずなんとかなるんじゃないかと思います．ただ，CUI で Unicode の文字は出せるのかな．出ないようなら，ssh サーバーを立てて，Mac などから接続するのがよいのではと思います．

**Comment by aminophen** [comment]
Solaris は VirtualBox 用 VM が Oracle 公式から落とせて，ほとんど何もせずに GNOME が使えるようになる，という感じです。 https://www.oracle.com/technetwork/jp/systems/hands-on-labs/s11-vbox-install-1408628-ja.html に従うだけでした。

**Comment by kmaed** [comment]
なるほど，楽ですね．

**Comment by trueroad** [comment]
1年以上前のissueなので既に解決されてしまっているのかもしれませんが、手元にCygwin 32 bit環境とTeX Live 2019 DVDがあるので、とりあえずインストーラを起動してみました。

```
======================> TeX Live installation procedure <=====================

======>   Letters/digits in <angle brackets> indicate   <=======
======>   menu items for actions or customizations      <=======

 Detected platform: Cygwin on Intel x86

*** WARNING: No binaries for your platform found.
 <B> set binary platforms: 0 out of 5

 <S> set installation scheme: scheme-full

 <C> set installation collections:
     40 collections out of 41, disk space required: 5698 MB

 <D> set directories:
   TEXDIR (the main TeX directory):
     /usr/local/texlive/2019
   TEXMFLOCAL (directory for site-wide local files):
     /usr/local/texlive/texmf-local
   TEXMFSYSVAR (directory for variable and automatically generated data):
     /usr/local/texlive/2019/texmf-var
   TEXMFSYSCONFIG (directory for local config):
     /usr/local/texlive/2019/texmf-config
   TEXMFVAR (personal directory for variable and automatically generated data):
     ~/.texlive2019/texmf-var
   TEXMFCONFIG (personal directory for local config):
     ~/.texlive2019/texmf-config
   TEXMFHOME (directory for user-specific files):
     ~/texmf

 <O> options:
   [ ] use letter size instead of A4 by default
   [X] allow execution of restricted list of programs via \write18
   [X] create all format files
   [X] install macro/font doc tree
   [X] install macro/font source tree
   [ ] create symlinks to standard directories
   [X] after install, set CTAN as source for package updates

 <V> set up for portable installation

Actions:
 <I> start installation to hard disk
 <P> save installation profile to 'texlive.profile' and exit
 <H> help
 <Q> quit

Enter command:
```

ということで、バイナリが無い旨の警告がでて、ひとつも選択されません。Cygwin 64 bitの環境で同様に起動すると、上記と違うところだけ抜粋しますが、

```
 Detected platform: Cygwin on x86_64

 <B> set binary platforms: 1 out of 5
```

という感じで警告が出ず、バイナリがひとつ選択されます。とりあえずこのまま32 bitの方に（普段使いの環境ではないので）最小インストールしてどうなるか見てみようと思います。

**Comment by trueroad** [comment]
全然ダメでした。いきなりインストール拒否されました。というわけでisoのインストーラはCygwin 32bit環境では使えないということのようです。

```
Actions:
 <I> start installation to hard disk
 <P> save installation profile to 'texlive.profile' and exit
 <H> help
 <Q> quit

Enter command: i
Installing to: /usr/local/texlive/2019

./install-tl: No binary platform specified/available, quitting.

$
```

**Comment by limaofu** [comment]
EdrawMath お試しましょう
https://math.edrawsoft.cn/
